server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -455,7 +459,10 @@ public synchronized IndexShard createShard(
         final Consumer<ShardId> globalCheckpointSyncer,
         final RetentionLeaseSyncer retentionLeaseSyncer,
         final SegmentReplicationCheckpointPublisher checkpointPublisher,
-        final RemoteStoreStatsTrackerFactory remoteStoreStatsTrackerFactory
+        final RemoteStoreStatsTrackerFactory remoteStoreStatsTrackerFactory,
+        final RepositoriesService repositoriesService,
+        final DiscoveryNode targetNode,
+        @Nullable DiscoveryNode sourceNode
```

Comment:
Is `targetNode` and `sourceNode` passed only to understand if this relocation is part of migration? Thinking if this can be inferred in the caller and passed as a flag?

Reply:
Yes, they are passed only to find out if target/source are remote nodes or not . We can pass two flags as well . But not sure if that would make it any better .

Reply:
Should we supply only the required information of whether the nodes are remote enabled or not?

Reply:
We also need to know if `source node != null` or not . So i am inclined to keep this as is .